src/com/mendix/widget/carousel/components/__tests__/Carousel.spec.ts

```diff
@@ -0,0 +1,113 @@
+import { ReactWrapper, ShallowWrapper, mount, shallow } from "enzyme";
+import { DOM, createElement } from "react";
+
+import { Carousel, CarouselProps, Image } from "../Carousel";
+import { CarouselItem } from "../CarouselItem";
+
+describe("Carousel", () => {
+    let images: Image[];
+    describe("shallow component", () => {
+        let output: ShallowWrapper<CarouselProps, any>;
+        let carouselWrapper: ShallowWrapper<CarouselProps, any>;
+        describe("with multiple images", () => {
+            beforeEach(() => {
+                images = [
+                    { url: "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" },
+                    { url: "https://www.google.com/images/nav_logo242.png" }
+                ];
+                output = shallow(createElement(Carousel, { images }));
+                carouselWrapper = output.find("div.carousel-inner");
+            });
+
+            it("renders a div of class carousel with a div child of class carousel-inner", () => {
+
+                expect(output).toMatchStructure(DOM.div({ className: "carousel" },
+                    DOM.div({ className: "carousel-inner" })));
+            });
+
+            it("renders a wrapper with the same number of children as the images", () => {
+
+                expect(carouselWrapper.children().length).toBe(images.length);
+            });
+
+            it("renders a wrapper with the first child active", () => {
+                const firstChild = carouselWrapper.children().first();
+
+                expect(firstChild.prop("active")).toBe(true);
+            });
+
+            it("renders a wrapper with only one active child", () => {
+                const activeChildren = carouselWrapper.children().filterWhere(c => c.prop("active"));
+
+                expect(activeChildren.length).toBe(1);
+            });
+        });
+
+        describe("with one image", () => {
+            beforeEach(() => {
+                images = [
+                    { url: "https://www.google.com/images/nav_logo242.png" }
+                ];
+                output = shallow(createElement(Carousel, { images }));
+                carouselWrapper = output.find("div.carousel-inner");
+            });
+
+            it("renders a wrapper with one child", () => {
+                const children = output.find("div.carousel-inner").children();
+
+                expect(children.length).toBe(1);
+            });
+        });
+
+        describe("with no images", () => {
+            beforeEach(() => {
+                output = shallow(createElement(Carousel));
+                carouselWrapper = output.find("div.carousel-inner");
+            });
+
+            it("renders a wrapper with no children", () => {
+                const children = output.find("div.carousel-inner").children();
+
+                expect(children.length).toBe(0);
+            });
+        });
+    });
+
+    describe("full component", () => {
```

Comment:
This shouldn't be a separate describe block, if you process the comments above than this should be gone and implemented in the existing tests.